packages/api/src/apps/api/app.ts

```diff
@@ -206,10 +208,10 @@ export default async (env: ProcessEnv) => {
     if (blockNumber - appState.lastBlockUpdate >= updateBlocks) {
       const end = profile("Updating state from block event");
       // update everyting
-      await services.registry(appState.lastBlock, blockNumber);
-      await services.lspCreator.update(appState.lastBlock, blockNumber);
-      await services.emps(appState.lastBlock, blockNumber);
-      await services.lsps.update(appState.lastBlock, blockNumber);
+      await services.registry(appState.lastBlockUpdate, blockNumber);
+      await services.lspCreator.update(appState.lastBlockUpdate, blockNumber);
+      await services.emps(appState.lastBlockUpdate, blockNumber);
+      await services.lsps.update(appState.lastBlockUpdate, blockNumber);
```

Comment:
these were using lastBlock, which updates every new block, but what we really want is the lastBlockUpdate as the start block, so we dont miss any events. lastBlockUpdate only changes when these services get updated.

Reply:
Ah tricky bug